app/views/projects/_form_0.html.erb

```diff
@@ -209,8 +209,19 @@
                          details:
                            t('projects.form_basics.license.details_html')
                        }) %>
-            <%= f.text_field :license,
+            <%=
+                # "Pattern" here is not a rigorous check of SPDX syntax.
+                # It only checks to ensure that only legal characters are
+                # allowed. Still, this can prevent some problems we've seen
+                # in some submissions, while permitting legal expressions like
+                # "(GPL-2.0+ WITH Bison-exception-2.2)".  See:
+                # https://spdx.org/spdx-specification-21-web-version
+                # Errors here do not cause *security* issues, so we're not
+                # too strict.
+                f.text_field :license,
                              class: 'form-control', hide_label: true,
+                             pattern: '(|[()A-Za-z0-9\-.+ ]*)',
```

Comment:
Yes, I know, I even documented that in the commit comment.  I don't think it's all that critical to implement more model validation in this case. It can't be a security vulnerability, *and* implementing serious license expression validation is way more effort (because it's a context-free grammar).

We can add it as a model validation, it's not too hard to add.  There is a complication: There are *existing* records that don't meet it (which is why I added this function). That's kind of a pain to deal with.

Because of the existing records, I think I'd rather start by checking just on the web page, and let people fix them as they go.  We can then add a model validator once a lot of the data has been fixed up.